### PR TITLE
Implement NamedNodeMap::getNamedItem*() (fixes #5454)

### LINKED ISSUE #5454: Implement named access to NamedNodeMap
Blocks #5395.

* Spec: <https://dom.spec.whatwg.org/#interface-namednodemap>.
* Code: `components/script/dom/namednodemap.rs`, `components/script/dom/webidls/NamedNodeMap.webidl`.
* Test: to be written.

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

File: components/script/dom/bindings/error.rs [hunk 1]
```diff
@@ -34,6 +34,8 @@ pub enum Error {
     InvalidCharacter,
     /// NotSupportedError DOMException
     NotSupported,
+    /// InUseAttributeError DOMException
+    InUseAttributeError,
```
Comment (Manishearth) [inline components/script/dom/bindings/error.rs]: Just `InUseAttribute`

**Comment by nox** [comment]
Blocked on #5503.

**Comment by Manishearth** [comment]
r=me with one nit, pending landing of #5503

**Comment by nox** [comment]
Rebased on master to get get_attribute_by_name(). The only thing remaining is tests in wpt to please @Ms2ger.

**Comment by Manishearth** [comment]
```
Unexpected Results
==================

/dom/interfaces.html
--------------------
PASS expected FAIL NamedNodeMap interface: operation getNamedItem(DOMString)
PASS expected FAIL NamedNodeMap interface: operation getNamedItemNS(DOMString,DOMString)
 9:13.96 LOG: MainThread INFO Closing logging queue
 9:13.96 LOG: MainThread INFO queue closed
```

**Comment by nox** [comment]
Sorry, forgot to run all the WPT tests.

**Comment by nox** [comment]
Amended with interfaces.html.ini fixed.

**Pull request merged**